### PR TITLE
[NTOS:IO] Implement IopAcquireFileObjectLock and use it to fix IopLockFileObject.

### PR DESCRIPTION
**Pull request by ThFabba**
JIRA issue: [CORE-14459](https://jira.reactos.org/browse/CORE-14459)

### CODE REVIEW

File: ntoskrnl/io/iomgr/file.c [hunk 1]
```diff
@@ -2123,6 +2131,52 @@ IopQueryAttributesFile(IN POBJECT_ATTRIBUTES ObjectAttributes,
...
+    _In_ PFILE_OBJECT FileObject,
+    _In_ KPROCESSOR_MODE WaitMode,
+    _In_ BOOLEAN Alertable,
+    _Out_ PBOOLEAN LockFailed)
+{
+    NTSTATUS Status;
+
+    PAGED_CODE();
+
+    InterlockedIncrement((PLONG)&FileObject->Waiters);
```
Comment (tkreuzer) [inline ntoskrnl/io/iomgr/file.c]: InterlockedIncrementUL

**Comment by HBelusca** [comment]
What's the status of this patch after commit https://github.com/reactos/reactos/commit/5472c1db82570a1ecd928c8c617320ee12ebfe15?

**Comment by HeisSpiter** [comment]
> What's the status of this patch after commit [5472c1d](https://github.com/reactos/reactos/commit/5472c1db82570a1ecd928c8c617320ee12ebfe15)?

It will have to be updated to handle the new case where the file object is locked.

**Comment by HeisSpiter** [comment]
Superseded with 8fbc488.
Thanks Thomas, works fine :-).

**Pull request closed without merging**